rasterio/_base.pyx

```diff
@@ -599,8 +609,100 @@ def window_shape(window, height=-1, width=-1):
     (a, b), (c, d) = eval_window(window, height, width)
     return b-a, d-c
 
+
 def window_index(window):
     return tuple(slice(*w) for w in window)
 
+
 def tastes_like_gdal(t):
     return t[2] == t[4] == 0.0 and t[1] > 0 and t[5] < 0
+
+
+cdef void *_osr_from_crs(object crs):
```

Comment:
@sgillies :+1: 
this is just what we need to use for detection of CRS mismatches in #268.  I'll work on adding a check based on this to rio-rasterize.